File: leet/array_string/increasing_triplet_subsequence_334.py
from typing import List
import math

class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        first = math.inf
        sec = math.inf

        for n in nums:
            if n <= first:
                first = n
            elif n <= sec:
                sec = n
            else:
                return True
        return False
        


# --------------------------------------------------------
# 是練習 if, else if, else 的好機會
# 要<= ，不然極端案例中，整個政列都是等大的，就會直接跳躍到第三個條件，回傳true

File: leet/array_string/increasing_triplet_subsequence_334.java
class Solution {
    public boolean increasingTriplet(int[] nums) {
        int first = Integer.MAX_VALUE;
        int sec = Integer.MAX_VALUE;
        for (int i : nums) {
            if (i < first) {
                first = i;
            } else if (i < sec) {
                sec = i;
            } else {
                return true;
            }
        }
        return false;
    }
}